source/adapters/level_zero/program.cpp

```diff
@@ -142,8 +153,10 @@ UR_APIEXPORT ur_result_t UR_APICALL urProgramBuild(
   ZeModuleDesc.pBuildFlags = Options;
   ZeModuleDesc.pConstants = Shim.ze();
 
-  ze_device_handle_t ZeDevice = Context->Devices[0]->ZeDevice;
+  ze_device_handle_t ZeDevice = phDevices[0]->ZeDevice;
   ze_context_handle_t ZeContext = Program->Context->ZeContext;
+  std::ignore = Context;
+  std::ignore = numDevices;
```

Comment:
wouldn't it be better to remove `Context` and `numDevices` in function definition? i.e. to have argument type but not name.

Reply:
> thanks @mmoadeli . I would prefer to leave them. That way, `urProgramBuild` and `urProgramBuildExp` are as compatible as possible, allowing us to see if in the future we can merge the two into a single API, or find other use cases.
> 
> For instance, if in `urProgramBuildExp` numDevices is 0, then the program could be built for all devices in the Context instead, which is kind of falling back again to urProgramBuild.
> 
> what do you think?
I did not mean to change the API, only changing it to below, to have the type for consistency and not to have the name to avoid using std::ignore.
```
UR_APIEXPORT ur_result_t UR_APICALL urProgramBuildExp(
    ur_context_handle_t, ///< [in] handle of the context instance.
    ur_program_handle_t Program, ///< [in] Handle of the program to build.
    uint32_t, ur_device_handle_t *phDevices,
    const char *Options ///< [in][optional] pointer to build options
                        ///< null-terminated string.
)```